### ISSUE TITLE
Documentation: Clarify expected arguments for queries.

### ISSUE THREAD

**Issue by alyti** [surrealdb/www.surrealdb.com#242]
### Description

Right now there's a large number of query types that accept queries only accept idents/literals, I've seen many users run into confusing parser errors/invalid queries due to this as it's not clearly noted anywhere.

Ideally these queries should at least support $variables, this way users don't have to open themselves to sql injections (by formatting their queries manually). 
In the interim it should at least be documented for beta.9 and maybe beta.10 (as it's still an issue on nightly).

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Ekwuno** [comment]
Hi @alyti thanks for pointing this out. Are you referring to SDKs or SurrealQL? Also if you have any links or examples that would be helpful as well.

**Comment by alyti** [comment]
I am referring to SurrealQL, notably almost all if not all queries in [DEFINE](https://surrealdb.com/docs/surrealql/statements/define) and [REMOVE](https://surrealdb.com/docs/surrealql/statements/remove) don't work with anything but simple literals.

**Comment by alyti** [comment]
Also [RELATE](https://surrealdb.com/docs/surrealql/statements/relate) to and from record could use some clarification on the accepted syntax, since I've also seen people try to use dot notation with variables inside (`RELATE $after.id->...->...`) and get confused too.

**Comment by kearfy** [comment]
I believe this issue came from a discussion in discord. There are generally some syntax "tricks" for which you need a deeper understanding of SurrealQL to know they even exist. In this case, Aly and theUno in Discord tried to create a relation from nested variables:
```surql
RELATE $nested.prop->...->...
```

The solution in this specific scenario is to wrap the variable around `()` brackets:
```surql
RELATE ($nested.prop)->...->...
```

As you can see here, it's not specific to relate statements, and not even this one specific "feature" (?) if I interpet this ticket correctly. I'm wondering when we'll be able to close it 😛

**Comment by alyti** [comment]
> As you can see here, it's not specific to relate statements, and not even this one specific "feature" (?) if I interpet this ticket correctly. I'm wondering when we'll be able to close it 😛

Yeah relate doc is just a side-quest, main issue is REMOVE/DEFINE queries only accepting literals, no tricks there this time 😛
